streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -123,6 +126,12 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
         timedOut
       }
 
+      // generate one more bacth to make sure RDD in lastJob is checkpointed.
+      if (!jobScheduler.receiverTracker.hasUnallocatedBlocks &&
+        ssc.graph.isCheckpointMissedLastTime) {
+        Thread.sleep(ssc.graph.batchDuration.milliseconds)
```

Comment:
If task execution time is longer than batchDuration, it usually means that the streaming application is not healhy and user need to adjust parameters to avoid such scenario.

And the "thread.sleep" here is to make sure a new job could be generated, it's nothing to do with job execution.